app/src/main/java/com/portfolio/tasky/entry/fragments/LoginFragment.kt

```diff
@@ -33,24 +35,85 @@ class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl() {
             openRegistrationFragment()
         }
     }
+    private fun setFieldValidations(loginFragment: LoginFragment?) {
+        clearValidatorParams()
+
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        etEmail.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail, LOGIN,
+                it
+            )
+        })
+
+        val etPassword = viewBinding.etPassword.subLayout.etInput
+        etPassword.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etPassword, LOGIN,
+                it
+            )
+        })
+    }
+
+    private fun removeFieldValidations(loginFragment: LoginFragment?) {
+        clearValidatorParams()
+
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        etEmail.removeTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail, LOGIN,
+                it
+            )
+        })
+
+        val etPassword = viewBinding.etPassword.subLayout.etInput
+        etPassword.removeTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etPassword, LOGIN,
+                it
+            )
+        })
+    }
+
+    override fun fieldsValidated(valid: Boolean) {
+        viewBinding.btnLogin.isEnabled = valid
+        valid.let {
+            if (it) {
+                viewBinding.etEmail.clearFocus()
+                viewBinding.etPassword.clearFocus()
+                viewBinding.btnLogin.requestFocus()
+            }
+        }
+
+    }
 
     private fun openRegistrationFragment() {
         (activity as EntryActivity).setTitle((activity as EntryActivity).getString(R.string.create_your_account))
 
-        setFragmentManager(activity?.supportFragmentManager!!)
+        setFragmentManager(activity?.supportFragmentManager as FragmentManager)
+        removeFragment(this)
+
         val registrationFragment = RegistrationFragment.getInstance()
         inflateFragment(registrationFragment, R.id.fragment_container)
     }
 
+    override fun onResume() {
+        super.onResume()
+        setFieldValidations(this)
+    }
+    override fun onDestroyView() {
+        super.onDestroyView()
+        removeFieldValidations(this)
+    }
+
     companion object {
+        private var loginFragment : LoginFragment? = null
         @JvmStatic
         fun getInstance(): LoginFragment {
-            val loginFragment = LoginFragment()
-            loginFragment.apply {
-                enterTransition = Slide(Gravity.BOTTOM)
-                exitTransition = Fade(Fade.MODE_OUT)
-            }
-            return loginFragment
+                if (loginFragment == null){
+                    loginFragment = LoginFragment()
+                }
+                loginFragment?.apply {
```

Comment:
LoginFragment won't be null here, since you asked if it's null before, and then created it

Reply:
Yes, I was trying to do some RnD and forgot to re-fix the code.

Done